### ISSUE TITLE
Parser truncates set element names containing hyphens in Table data

### ISSUE THREAD

**Issue by jeffreyhorn** [jeffreyhorn/nlp2mcp#665]
## Summary

The GAMS parser truncates set element names that contain hyphens when parsing Table data. For example, `20-bond-wt` becomes `20` in the parsed parameter data, but is correctly parsed as `20-` in the set declaration (though still missing `bond-wt`).

## Reproduction

**Model:** `data/gamslib/raw/ajax.gms`

**Original GAMS source:**
```gams
Set
   g 'paper grades'      / 20-bond-wt, 25-bond-wt, c-bond-ext, tissue-wrp /;

Table dempr(g,*) 'demand and prices'
                  demand  price
   20-bond-wt      30000     77
   25-bond-wt      20000     81
   c-bond-ext      12000     99
   tissue-wrp       8000    105;
```

**Parsed IR:**
```python
# Set g members - note truncation to '20-' instead of '20-bond-wt'
Set g members: ['20-', '25-', 'c-bond-ext', 'tissue-wrp']

# Parameter dempr values - note '20' instead of '20-' or '20-bond-wt'
dempr values:
  ('20', 'demand') = 30000.0
  ('20', 'price') = 77.0
  ('25', 'demand') = 20000.0
  ('25', 'price') = 81.0
  ("'c-bond-ext'", 'demand') = 12000.0
  ...
```

## Impact

1. **Set/Parameter mismatch**: Set `g` has element `'20-'` but parameter `dempr` references element `20` - these don't match
2. **Full name lost**: The complete element name `20-bond-wt` is lost entirely
3. **Quoted elements preserved**: Elements like `c-bond-ext` that get quoted are parsed correctly (as `'c-bond-ext'`)

## Root Cause

The parser appears to be treating `-` as a token separator in Table row labels, splitting `20-bond-wt` into separate tokens. The set declaration parsing handles this differently (preserving `20-`) but still truncates.

## Expected Behavior

- Set `g` should have members: `['20-bond-wt', '25-bond-wt', 'c-bond-ext', 'tissue-wrp']`
- Parameter `dempr` keys should be: `('20-bond-wt', 'demand')`, etc.

## Related

- Discovered during PR #664 review
- Similar issue may affect other models with hyphenated element names

**Comment by jeffreyhorn** [comment]
Fixed. Extended preprocessor pattern to match number-starting hyphenated identifiers, improved column matching with range-based algorithm, and updated grammar to allow STRING in dotted_label. Added regression tests.
